### PR TITLE
Improve and fix utf-8 encoding / decoding for module

### PR DESCRIPTION
**Pull request by MaxGraey**
See [this](https://github.com/AssemblyScript/assemblyscript/pull/1341#discussion_r440895843) comment for details
- [x] I've read the contributing guidelines

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -2402,26 +2386,18 @@ export function readString(ptr: usize): string | null {
...
-        if ((cp & 0xFC) == 0xF8) {
-          cp = ((cp & 3) << 24) | (u1 << 18) | (u2 << 12) | (u3 << 6) | u4;
-        } else {
-          u5 = binaryen.__i32_load8_u(ptr++) & 63;
-          cp = ((cp & 1) << 30) | (u1 << 24) | (u2 << 18) | (u3 << 12) | (u4 << 6) | u5;
-        }
+        assert("Invalid UTF8 sequence during readString");
       }
     }
-    arr.push(cp);
```
Comment (dcodeIO) [inline src/module.ts:2414]: The important part for the fix appears to be removing this line, that correct? If so, can we make this PR just remove the line as a "fix", before merging the larger one as "refactor"?

Comment (MaxGraey) [reply src/module.ts:2414]: It will be great if this PR merged first. Then I just sync PR with refactoring with master

File: src/module.ts [hunk 2]
```diff
@@ -2402,26 +2386,18 @@ export function readString(ptr: usize): string | null {
...
         cp = ((cp & 7) << 18) | (u1 << 12) | (u2 << 6) | u3;
       } else {
-        u4 = binaryen.__i32_load8_u(ptr++) & 63;
-        if ((cp & 0xFC) == 0xF8) {
-          cp = ((cp & 3) << 24) | (u1 << 18) | (u2 << 12) | (u3 << 6) | u4;
-        } else {
-          u5 = binaryen.__i32_load8_u(ptr++) & 63;
-          cp = ((cp & 1) << 30) | (u1 << 24) | (u2 << 18) | (u3 << 12) | (u4 << 6) | u5;
-        }
+        assert("Invalid UTF8 sequence during readString");
```
Comment (dcodeIO) [inline src/module.ts:2389]: Should be `assert(false, optionalString)` so the string is eliminated if `--noAssert` is set

Comment (MaxGraey) [reply src/module.ts:2389]: good catch!

**Comment by dcodeIO** [comment]
For some reason this PR didn't make it to the changelog, and there's nothing in the [log](https://github.com/AssemblyScript/assemblyscript/runs/778640011?check_suite_focus=true#step:7:32) indicating why. Perhaps due to `refactor(compiler):`?

**Comment by MaxGraey** [comment]
I guess bot ignore `refactor:` and `refactor(scpe):`

**Pull request merged**